nexus/src/app/sagas/instance_create.rs

```diff
@@ -187,9 +187,12 @@ impl NexusSaga for SagaInstanceCreate {
             Ok(())
         }
 
-        for (i, group) in
-            params.create_params.anti_affinity_groups.iter().enumerate()
+        let aa_groups = match params.create_params.anti_affinity_groups.as_ref()
         {
+            Some(groups) => groups,
+            None => &Vec::new(),
+        };
+        for (i, group) in aa_groups.iter().enumerate() {
```

Comment:
take it or leave it: I believe this could be simplified to:
```suggestion
        let aa_groups = params.create_params.anti_affinity_groups.iter().flat_map(Vec::iter);
        for (i, group) in aa_groups.enumerate() {
```
(rustfmt will probably want to wrap this line)